recipe/meta.yaml

```
@@ -1,47 +1,96 @@
 {% set name = "testtools" %}
-{% set version = "2.4.0" %}
-{% set bundle = "tar.gz" %}
-{% set hash_type = "sha256" %}
-{% set hash = "64c974a6cca4385d05f4bbfa2deca1c39ce88ede31c3448bee86a7259a9a61c8" %}
-{% set build = 0 %}
+{% set version = "2.7.2" %}
 
 package:
-  name: {{ name }}
+  name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.{{ bundle }}
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ bundle }}
-  {{ hash_type }}: {{ hash }}
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/testtools-{{ version }}.tar.gz
+  sha256: 5be5bbc1f0fa0f8b60aca6ceec07845d41d0c475cf445bfadb4d2c45ec397ea3
 
 build:
-  number: {{ build }}
-  script: python setup.py install --single-version-externally-managed --record=record.txt
+  number: 0
+  skip: true  # [py<38]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vv
 
 requirements:
   host:
+    - pip
     - python
-    - setuptools
-    - pbr >=0.11
-
+    - setuptools >=61
+    - hatchling
+    - hatch-vcs
+    - wheel
   run:
     - python
-    - pbr >=0.11
+  run_constrained:
+    - fixtures >=2.0
+    - twisted >=22.2
+
+# =======================
+# Test configuration
+# =======================
+{% set test_root = "testtools/tests" %}
+
+# - Skip twisted
+# - Skip fixtures
+# - Others: assertion/runtime error
+{% set skip_files = [
+    "testtools/tests/twistedsupport",
+    "testtools/tests/matchers",
+    "testtools/tests/test_testcase.py",
+    "testtools/tests/test_testresult.py",
+    "testtools/tests/test_testsuite.py"
+] %}
+
+# [win-64]
+# with open(path, "rb") as stream:
+# PermissionError: [Errno 13] Permission denied: 'C:\\cygwin64\\tmp\\tmpkbmxjt49'
+{% set skip_keywords = [
+    "test_from_file_with_simple_seek",
+    "test_from_file_with_whence_seek"
+] %}
 
 test:
+  requires:
+    - pip
+    - pytest
+  source_files:
+    - {{ test_root }}
   imports:
     - testtools
+    - testtools.assertions
+    - testtools.compat
+    - testtools.content
+    - testtools.content_type
+    - testtools.helpers
+    - testtools.matchers
+    - testtools.monkey
+    - testtools.runtest
+    - testtools.tags
+  commands:
+    - pip check
+    - python -c "import testtools; print(testtools.__version__)"
```

This line does not test anything. Just print the version?
Maybe you want to dosomething like that:
```suggestion
    - python -c "from importlib.metadata import version; assert(version('{{ name }}')=='{{ version }}')"

```